### PR TITLE
Add initial commit for new sample upload flow.

### PR DESCRIPTION
**Pull request by MarkAZhang**
Added new endpoints that upload metadata together with the sample.
Will deprecate old endpoints after conversion happens.

Also added endpoint that validates metadata for new samples.

Added components for each page of the new upload flow, though styling is TBD.

Everything is admin-only for now.

Also added Rails Integration tests for new endpoints.

---

UI and styling will be polished in a future commit, but here are screenshots of the flow for context. Everything is admin-only.

screen shot 2019-02-06 at 7 13 27 pm

screen shot 2019-02-06 at 7 13 53 pm

screen shot 2019-02-06 at 7 13 59 pm

screen shot 2019-02-06 at 7 14 04 pm

screen shot 2019-02-06 at 7 16 54 pm

### CODE REVIEW

File: app/assets/src/components/views/SampleUploadFlow/SampleUploadFlow.jsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+          name: sampleName,
+          host_genome_id: this.state.hostId
+        })
+      );
+    }
+
+    return {};
+  };
+
+  renderStep = () => {
```
Comment (jshoe) [inline app/assets/src/components/views/SampleUploadFlow/SampleUploadFlow.jsx]: could use a switch statement

File: app/controllers/samples_controller.rb [hunk 2]
```diff
@@ -181,6 +183,31 @@ def bulk_upload
...
+    # Ignore samples that aren't part of editable projects.
+    # TODO(mark): Show error for samples that aren't part of project.
+    samples_to_upload = samples_to_upload.select { |sample| editable_project_ids.include?(Integer(sample["project_id"])) }
+
+    errors, samples = upload_samples_with_metadata(samples_to_upload, metadata).values_at("errors", "samples")
+
+    respond_to do |format|
+      if errors.empty? && !samples.empty?
+        # tags = %W[client:web type:bulk user_id:#{current_user.id}]
+        # MetricUtil.put_metric_now("samples.created", @samples.count, tags)
```
Comment (jshoe) [inline app/controllers/samples_controller.rb]: ah you can leave these in for now until we get the new dashboards all set up

File: app/controllers/samples_controller.rb [hunk 3]
```diff
@@ -588,12 +615,19 @@ def results_folder
...
   def new
     @sample = nil
     @projects = current_power.updatable_projects
     @host_genomes = host_genomes_list ? host_genomes_list : nil
   end
 
+  # GET /samples/upload
+  def upload
+    @projects = current_power.updatable_projects
+    @host_genomes = host_genomes_list ? host_genomes_list : nil
```
Comment (jshoe) [inline app/controllers/samples_controller.rb]: could just be `host_genomes_list || nil` ?

File: app/controllers/samples_controller.rb (hunk 2 above)
Comment (MarkAZhang) [reply app/controllers/samples_controller.rb]: Oops! I didn't mean to remove this.

**Pull request merged**